modules/messages/src/lib.rs

```diff
@@ -746,23 +688,41 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I> {
 		fn build(&self) {
-			PalletOperatingMode::<T, I>::put(&self.operating_mode);
+			PalletOperatingMode::<T, I>::put(self.operating_mode);
 			if let Some(ref owner) = self.owner {
 				PalletOwner::<T, I>::put(owner);
 			}
 		}
 	}
 
+	// TODO: Maybe useless for Darwinia.
 	impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		/// Get stored data of the outbound message with given nonce.
 		pub fn outbound_message_data(
 			lane: LaneId,
 			nonce: MessageNonce,
 		) -> Option<MessageData<T::OutboundMessageFee>> {
-			OutboundMessages::<T, I>::get(MessageKey { lane_id: lane, nonce })
+			OutboundMessages::<T, I>::get(MessageKey { lane_id: lane, nonce }).map(Into::into)
+		}
+
+		/// Prepare data, related to given inbound message.
+		pub fn inbound_message_data(
```

Comment:
`inbound_message_data`  seems useless to us.

Reply:
@fewensa Could you check this?

Reply:
Yes, we don't need this.
https://github.com/paritytech/parity-bridges-common/blob/f924be9867bf381c76f1860b60aac4d6d4c58b24/bin/runtime-common/src/messages_api.rs#L64
this is supported by message api. our bridge read storage directly.
additional the `outbound_message_data` is also can be clean. 
https://github.com/paritytech/parity-bridges-common/blob/f924be9867bf381c76f1860b60aac4d6d4c58b24/bin/runtime-common/src/messages_api.rs#L37